src/Todo/Model.hs

```diff
@@ -32,3 +42,58 @@ data Todo = Todo
     , todoDeadline    :: Maybe Deadline
     , todoDescription :: Maybe String
     } deriving (Eq, Show)
+
+data TodoDB = TodoDB 
+    { tododbTodos :: Todos
+    , tododbTags  :: [Tag]
+    } deriving (Eq, Show)
+
+
+addTodo :: Todo -> TodoDB -> TodoDB
+addTodo todo TodoDB { .. } = TodoDB { tododbTodos = todo : tododbTodos, .. }
+
+addTag :: Tag -> TodoDB -> TodoDB
+addTag tag TodoDB { .. } = TodoDB { tododbTags = tag : tododbTags, .. }
+
+deleteTag :: Tag -> TodoDB -> TodoDB
+deleteTag tag db@TodoDB { .. } = case List.find ((tag :| [] ==) . todoTags) tododbTodos of 
+    Just _  -> db
```

Comment:
Btw, I don't understand logic here. You don't delete tag if there're exist TODO items with only this single `tag`? I think in that case the return type of your function should be `Either DBError TodoDB`.